### ISSUE TITLE
Cannot use stack lts-18.6: "Failed to get project GHC version"

### ISSUE THREAD

**Issue by sweirich** [haskell/haskell-language-server#2224]
### Your environment
Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:

```sh
sweirich@SEASNet-50-03 02-trees % "/Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin" --probe-tools
haskell-language-server version: 1.4.0.0 (GHC: 8.10.4) (PATH: /Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin) (GIT hash: 253547816ee216c53ee7dacc0ad3cac43e863d30)
Tool versions found on the $PATH
cabal:		3.2.0.0
stack:		2.7.3
ghc:		8.10.4


```
Which OS do you use:
MacOS
Which lsp-client do you use:
VS Code
Describe your project (alternative: link to the project):
https://github.com/cis552/02-trees
Contents of `hie.yaml`:
```yaml
cradle:
  stack:
    stackYaml: "./stack.yaml"
```

### Steps to reproduce
stack.yaml must be lts-8.6.   Other recent lts work fine. (Tested 18.8, 18.10, 17.2, 16.31)

### Actual behaviour
Couldn't figure out what GHC version the project is using:

/Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin --project-ghc-version exited with exit code 1: Found "/Users/sweirich/552/lecture-repos/02-trees/hie.yaml" for "/Users/sweirich/552/lecture-repos/02-trees/a" Failed to get project GHC version:CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack --stack-yaml /Users/sweirich/552/lecture-repos/02-trees/./stack.yaml setup --silent","",""]}

### Include debug information
Execute in the root of your project the command `haskell-language-server --debug .` and paste the logs here:

<details>
<summary>
Debug output:
</summary>

```
sweirich@SEASNet-50-03 02-trees % "/Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin" --debug      
Found "/Users/sweirich/552/lecture-repos/02-trees/hie.yaml" for "/Users/sweirich/552/lecture-repos/02-trees/a"
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper-1.4.0-darwin) Version 1.4.0.0, Git revision 253547816ee216c53ee7dacc0ad3cac43e863d30 (dirty) x86_64 ghc-8.10.4
Current directory: /Users/sweirich/552/lecture-repos/02-trees
Operating system: darwin
Arguments: ["--debug"]
Cradle directory: /Users/sweirich/552/lecture-repos/02-trees
Cradle type: Stack

Tool versions found on the $PATH
cabal:		3.2.0.0
stack:		2.7.3
ghc:		8.10.4


Consulting the cradle to get project GHC version...
Failed to get project GHC version:CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack --stack-yaml /Users/sweirich/552/lecture-repos/02-trees/./stack.yaml setup --silent","",""]}
```
</details>

Paste the logs from the lsp-client, e.g. for [VS Code](https://github.com/haskell/vscode-haskell#troubleshooting)

<details>
<summary>
LSP logs:
</summary>

```
[client][INFO] Searching for server executables haskell-language-server-wrapper,haskell-language-server in $PATH
[client][INFO] Downloading haskell-language-server
[client][INFO] Fetching the latest release from GitHub or from cache
[client][INFO] The latest release is 1.4.0
[client][INFO] Figure out the ghc version to use or advertise an installation link for missing components
[client][INFO] Working out the project GHC version. This might take a while...
[client][INFO] Executing '/Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin --project-ghc-version' in cwd '/Users/sweirich/552/lecture-repos/02-trees' to get the project or file ghc version
[client][INFO] Execution of '/Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin --project-ghc-version' terminated with code 1
[client][ERROR] Error executing '/Users/sweirich/Library/Application Support/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-darwin --project-ghc-version' with error code 1
[client][ERROR] stderr: Found "/Users/sweirich/552/lecture-repos/02-trees/hie.yaml" for "/Users/sweirich/552/lecture-repos/02-trees/a"
Failed to get project GHC version:CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack --stack-yaml /Users/sweirich/552/lecture-repos/02-trees/./stack.yaml setup --silent","",""]}
```
</details>

**Comment by sweirich** [comment]
```
sweirich@SEASNet-50-03 02-trees % stack --stack-yaml /Users/sweirich/552/lecture-repos/02-trees/./stack.yaml setup
The GHC located at /Users/sweirich/.stack/programs/x86_64-osx/ghc-8.10.4/bin/ghc-8.10.4 failed to compile a sanity check. Please see:

    http://docs.haskellstack.org/en/stable/install_and_upgrade/

for more information. Exception was:
Received ExitFailure 1 when running
Raw command: /Users/sweirich/.stack/programs/x86_64-osx/ghc-8.10.4/bin/ghc-8.10.4 /private/var/folders/p3/pkythxvx6rq9q054797y4bb80000gn/T/stack-sanity-check-cbcf3f5295538e79/Main.hs -no-user-package-db
Run from: /private/var/folders/p3/pkythxvx6rq9q054797y4bb80000gn/T/stack-sanity-check-cbcf3f5295538e79/
Standard output:

Loaded package environment from /Users/sweirich/.ghc/x86_64-darwin-8.10.4/environments/default
[1 of 1] Compiling Main             ( /private/var/folders/p3/pkythxvx6rq9q054797y4bb80000gn/T/stack-sanity-check-cbcf3f5295538e79/Main.hs, /private/var/folders/p3/pkythxvx6rq9q054797y4bb80000gn/T/stack-sanity-check-cbcf3f5295538e79/Main.o )
Standard error:


/private/var/folders/p3/pkythxvx6rq9q054797y4bb80000gn/T/stack-sanity-check-cbcf3f5295538e79/Main.hs:1:1: error:
    Could not load module âDistribution.Simpleâ
    It is a member of the hidden package âCabal-3.2.1.0â.
    You can run â:set -package Cabalâ to expose it.
    (Note: this unloads all the modules in the current scope.)
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
  |
1 | import Distribution.Simple
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^

```

**Comment by sweirich** [comment]
```
sweirich@SEASNet-50-03 02-trees % stack --stack-yaml /Users/sweirich/552/lecture-repos/02-trees/./stack.yaml setup --silent
sweirich@SEASNet-50-03 02-trees % 
```

**Comment by sweirich** [comment]
I added the hie.yaml in trying to track down the issue. The issue was present before I added this file.

**Comment by jneira** [comment]
what about if you delete `/Users/sweirich/.ghc/x86_64-darwin-8.10.4/environments/default`?, maybe it is interfering with the build doing by stack to make the "sanity check"

**Comment by sweirich** [comment]
That fixed it. Thanks!

**Comment by jneira** [comment]
nice! so I think we can close this, the stack setup --silent thing can be discussed in the hie-bios mentioned issue

**Comment by dnovick** [comment]
So, I'm not sure if I should open a new ticket or not. If I need to, let me know. That said...

I just started having exactly the same issue with lts18.10 (though it happened with lts18.5 too), and as far as I know, nothing significant changed. I tried a number of things, including:
- Deleting all Haskell language server instances under .config/Code/User/haskell.haskell.
- Uninstalling/reinstalling the Haskell extension in VS Code (including removing the haskell.haskell directory)
- Creating a new stack based Haskell project and opening it as a workspace in VS Code (The same error occurs). 
- Ran the following commands from a terminal in my project root (both succeeded):
-      /home/dnovick/.config/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-linux --project-ghc-version
-    /home/dnovick/.config/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-linux --probe-tools

Here's the error message I get:

[client][ERROR] Error executing '/home/dnovick/.config/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-linux --project-ghc-version' with error code 1
[client][ERROR] stderr: No 'hie.yaml' found. Try to discover the project type!
Failed to get project GHC version:CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack setup --silent","",""]}

Here's the output to the 2 manual calls to the HLS wrapper:

--project-ghc-version: 
No 'hie.yaml' found. Try to discover the project type!
8.10.7

--probe-tools

haskell-language-server version: 1.4.0.0 (GHC: 8.10.4) (PATH: /home/dnovick/.config/Code/User/globalStorage/haskell.haskell/haskell-language-server-wrapper-1.4.0-linux) (GIT hash: 253547816ee216c53ee7dacc0ad3cac43e863d30)
Tool versions found on the $PATH
cabal:		3.0.0.0
stack:		2.7.3
ghc:		8.8.4


I also ran the HLS wrapper with no command line parameters and received the following. I thought the error regarding the missing HLS executable might be significant.:

No 'hie.yaml' found. Try to discover the project type!
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper-1.4.0-linux) Version 1.4.0.0, Git revision 253547816ee216c53ee7dacc0ad3cac43e863d30 (dirty) x86_64 ghc-8.10.4
Current directory: /home/dnovick/gitrepos/haskell/key-management
Operating system: linux
Arguments: []
Cradle directory: /home/dnovick/gitrepos/haskell/key-management
Cradle type: Stack

Tool versions found on the $PATH
cabal:		3.0.0.0
stack:		2.7.3
ghc:		8.8.4


Consulting the cradle to get project GHC version...
Project GHC version: 8.10.7
haskell-language-server exe candidates: ["haskell-language-server-8.10.7","haskell-language-server"]
Cannot find any haskell-language-server exe, looked for: haskell-language-server-8.10.7, haskell-language-server

I tried to understand from the previous info on this issue what I should do to fix the issue, but didn't see an obvious solution.
